lib/Base/Tensor.cpp

```diff
@@ -379,6 +380,60 @@ std::string Tensor::toString(unsigned maxNumElem) const {
   return os.str();
 }
 
+void Tensor::dumpToRawBinaryFile(const char *filename) {
+  std::ofstream fs;
+  fs.open(filename, std::ios::out | std::ios::binary);
+  DCHECK(fs.is_open()) << "Failed to open file: " << filename;
+  auto handle = getHandle<>();
+  for (dim_t rdi = 0; rdi < actualSize(); rdi++) {
+    fs.write((char *)&handle.raw(rdi), sizeof(handle.raw(rdi)));
+  }
+  fs.close();
+}
+
+void Tensor::dumpToRawTextFile(const char *filename) {
+  std::ofstream fs;
+  fs.open(filename);
+  DCHECK(fs.is_open()) << "Failed to open file: " << filename;
+  auto handle = getHandle<>();
+  for (dim_t rdi = 0; rdi < actualSize(); rdi++) {
+    fs << handle.raw(rdi) << ", ";
+  }
+  fs.close();
+}
+
+void Tensor::loadFromRawBinaryFile(const char *filename) {
```

Comment:
What happens in the loading methods if the file being read contains less elements (or more elements) than expected? It seems like it should be possible to detect such errors.

Reply:
Added verification: when the binary file is loaded, the file size is checked against the tensor size.